### PR TITLE
[PROTOCOL-119] Adding Interest Validator

### PR DESCRIPTION
**Pull request by salazarguille**
It includes:

- A new InterestValidator interface to be used in the LendingPool.withdrawInterest function. By default, it will be empty. So we will not be validating the amount to withdraw. But it allows us to implement a new version in the future.
- New unit tests

image

### CODE REVIEW

File: contracts/base/Lenders.sol [hunk 1]
```diff
@@ -155,6 +129,32 @@ contract Lenders is Base, LendersInterface {
...
+        lendingPoolAddress.requireNotEmpty("LENDING_POOL_MUST_BE_PROVIDED");
+        interestConsensusAddress.requireNotEmpty("CONSENSUS_MUST_BE_PROVIDED");
+
+        _initialize(settingAddress, marketsAddress);
+
+        zToken = zTokenAddress;
+        lendingPool = lendingPoolAddress;
+        interestConsensus = InterestConsensusInterface(interestConsensusAddress);
+    }
+
```
Comment (passabilities) [inline contracts/base/Lenders.sol:157]: Any reason this was moved down here? Makes more sense to keep it towards the top.

File: contracts/base/LendingPool.sol [hunk 2]
```diff
@@ -238,8 +216,76 @@ contract LendingPool is Base, LendingPoolInterface {
...
+        _initialize(settingsAddress, marketsAddress);
+
+        zToken = ZTokenInterface(zTokenAddress);
+        lendingToken = IERC20(lendingTokenAddress);
+        lenders = LendersInterface(lendersAddress);
+        loans = loansAddress;
+        cToken = cTokenAddress;
+        interestValidator = InterestValidatorInterface(interestValidatorAddress);
+    }
+
```
Comment (passabilities) [inline contracts/base/LendingPool.sol:282]: Again should leave the `initializer` towards the top.

File: contracts/base/Lenders.sol (hunk 1 above)
Comment (salazarguille) [reply contracts/base/Lenders.sol:157]: Yes, this is due to gas consumption reasons. The functions that will be invoked more frequently should be on the top. So they consume less gas when EVM compares the sig function.

File: contracts/base/LendingPool.sol (hunk 2 above)
Comment (salazarguille) [reply contracts/base/LendingPool.sol:282]: The same reason.

**Pull request merged**